others/postfix_evaluation.cpp

```diff
@@ -0,0 +1,159 @@
+ * When the expression is ended, the number in the stack is the final answer
+ */
+#include <array>           // for std::array
+#include <iostream>        // for io operations 
+#include <cassert>  // for assert
+
+// Maximum size for stack
+const int MAX_SIZE = 20;
+
+// Defining stack
+std::array<int, MAX_SIZE> stack;
+
+// Index of top value in stack array
+int top;
+
+/**
+ * Pushing operand, also called the integer in the array to the stack
+ * @param operand integer value from the input array or evaluation
+ * @returns none
+ */
+void push(int operand){
+    top++;
+    stack[top]=operand;
+}
+
+/**
+ * Popping operand, also called the integer from the stack
+ * @param none
+ * @returns operand integer on top of stack
+ */
+int pop(){
+    int operand = stack[top];
+    top--;
+    return operand;
+}
+
+/**
+ * Evaluate answer using given operands and operation
+ * @param a integer
+ * @param b integer
+ * @param operation operation to be performed with respective integers
+ * @returns none
+ */  
+void evaluate(int a, int b, char operation){
+    int c;
+    switch(operation){
+        case '+':
+            c=a+b;
+            push(c);
+            break;
+
+        case '-':
+            c=a-b;
+            push(c);
+            break;
+
+        case '*':
+            c=a*b;
+            push(c);
+            break;
+
+        case '/':
+            c=a/b;
+            push(c);
+            break;
+
+        default:
+            std::cout<<"Operator not defined\n";
+            break;
+    }
+}
+
+
+/**
+ * Postfix Evaluation algorithm to compute the value from given input array
+ * @param input Array of characters consisting of integers and operations
+ * @param N Array size
+ * @returns stack[top] returns the top value from the stack
+ */ 
+template <std::size_t N>
+int postfix_evaluation(std::array<char,N> input){
+
+    top = -1;
+    int j=0;
+    while(input[j]){
+        char scan = input[j];
+        if(isdigit(scan)){
+            
+            push(scan-'0');
+            
+
+        }else{
+            int op2 = pop();
+            int op1 = pop();
+
+            evaluate(op1,op2,scan);
+        }
+        j++;
+    }
+
+    std::cout<<stack[top]<<"\n";
+
+    return stack[top];
+
+}
+
+/**
+ * Test function 1 with input array
+ * {'2', '3', '1', '*', '+', '9', '-'}
+ * @returns none
+ */
+static void test_function_1(){
+
+    std::array<char,7> input = {'2', '3', '1', '*', '+', '9', '-'};
+
+    int answer = postfix_evaluation(input);
+
+    assert(answer==-4);
+
+}
+
+/**
+ * Test function 2 with input array
+ * {'1', '2', '+', '2', '/', '5', '*', '7', '+'}
+ * @returns none
+ */
+static void test_function_2(){
+        
+    std::array<char, 9> input = {'1', '2', '+', '2', '/', '5', '*', '7', '+'};
+    int answer = postfix_evaluation(input);
+
+    assert(answer==6);
+
+}
+
+/**
+ * Main function
+ */
+int main(){
+
+    test_function_1();
+    test_function_2();
+
+    return 0;
+}
+
+
+
```

Comment:
Overall, it seems good. One more suggestion I would give is that maybe it would be better to create a custom class/structure to make most of these postfix evaluations to be applied more generally (not just integers), but take this with a pinch of salt as it would require both operands to be type casted.

Reply:
I can try to change it to float instead of integers to see if it still works fine, if you like.

Reply:
I can try to change it to float instead of integers if you want. I have only seen cases where it's applied to integers.

Reply:
Okay, float works too - for learning purposes it's good. However your loop is parsing numbers char by char which would not be able to parse floats properly (your test cases would fail if you use floating point numbers in any one of them)

The reason I asked for template types was that sometime we work with general types, for example let's say I have made a class of field (take complex numbers as an example), which has the operations +, -, * and / defined, it has all the operations but this program won't be able to adapt to that specific field (complex numbers). To make it work, we would need to change the stacks type and functions' input types and check whether this works without giving errors (given that we can parse complex numbers from strings without issue - easy to do if you rewrite the while loop such that it reads the whole number instead of character by character which you would need to do for floats anyways). All this even though there is nothing inherently different in the program as long as we are working in a field.

So yeah, don't change it if you prefer it the way it is because the purpose of this repository is to teach newcomers algorithms (I'm also one of them) and this is a very nice algorithm at that😄. After that people should be able to adapt to their preferences and open PRs with improvements.

PS: sorry if I have missed something, I've been busy with exams and I'm reviewing this on my phone😂

Reply:
Already made the necessary changes with respect to float. If no further changes to be done, you can approve it for the PR to be merged